tests/store/artifact/test_s3_artifact_repo.py

```diff
@@ -349,3 +352,67 @@ def test_delete_artifacts(s3_artifact_root, tmp_path):
     repo.delete_artifacts()
     tmpdir_objects = repo.list_artifacts()
     assert not tmpdir_objects
+
+
+def test_create_multipart_upload(s3_artifact_root):
+    repo = get_artifact_repository(posixpath.join(s3_artifact_root, "some/path"))
+    create = repo.create_multipart_upload("local_file")
+
+    # confirm that a mpu is created with the correct upload_id
+    bucket, _ = repo.parse_s3_compliant_uri(s3_artifact_root)
+    s3_client = repo._get_s3_client()
+    response = s3_client.list_multipart_uploads(Bucket=bucket)
+    uploads = response.get("Uploads")
+    assert len(uploads) == 1
+    assert uploads[0]["UploadId"] == create.upload_id
+
+
+def test_complete_multipart_upload(s3_artifact_root):
+    repo = get_artifact_repository(posixpath.join(s3_artifact_root, "some/path"))
+    local_file = "local_file"
+    create = repo.create_multipart_upload(local_file, 2)
+
+    # cannot complete invalid upload
+    fake_parts = [
+        MultipartUploadPart(part_number=1, etag="fake_etag1"),
+        MultipartUploadPart(part_number=2, etag="fake_etag2"),
+    ]
+    with pytest.raises(botocore.exceptions.ClientError, match=r".+"):
+        repo.complete_multipart_upload(local_file, create.upload_id, fake_parts)
+
+    # can complete valid upload
+    parts = []
+    data = b"0" * 5 * 1024 * 1024
+    for credential in create.credentials:
+        url = credential.url
+        response = requests.put(url, data=data)
+        parts.append(
+            MultipartUploadPart(part_number=credential.part_number, etag=response.headers["ETag"])
+        )
+
+    repo.complete_multipart_upload(local_file, create.upload_id, parts)
+
+    # verify upload is completed
+    bucket, _ = repo.parse_s3_compliant_uri(s3_artifact_root)
+    s3_client = repo._get_s3_client()
+    response = s3_client.list_multipart_uploads(Bucket=bucket)
+    assert not response.get("Uploads")
+
+
+def test_abort_multipart_upload(s3_artifact_root):
+    repo = get_artifact_repository(posixpath.join(s3_artifact_root, "some/path"))
+    local_file = "local_file"
+    create = repo.create_multipart_upload(local_file, 2)
+
+    # cannot abort a non-existing upload
+    with pytest.raises(botocore.exceptions.ClientError, match=r".+"):
+        repo.abort_multipart_upload(local_file, "fake_upload_id")
+
+    # can abort the created upload
+    repo.abort_multipart_upload(local_file, create.upload_id)
+
+    # verify upload is aborted
+    bucket, _ = repo.parse_s3_compliant_uri(s3_artifact_root)
+    s3_client = repo._get_s3_client()
+    response = s3_client.list_multipart_uploads(Bucket=bucket)
+    assert not response.get("Uploads")
```

Comment:
What does `response.get("Uploads")` return? None?

Reply:
yes it's None

Reply:
Got it. Can we use `assert response.get("Uploads") is None` to clarify what we exepct here?

Reply:
sure, i updated it

Reply:
thanks for the update!